packages/make-styles-webpack-loader/src/webpackLoader.test.ts

```diff
@@ -0,0 +1,168 @@
+    {
+      context: __dirname,
+      entry: entryPath,
+
+      mode: 'development',
+
+      output: {
+        path: path.resolve(__dirname),
+        filename: 'bundle.js',
+      },
+
+      module: {
+        rules: [
+          {
+            test: /\.(js|tsx|ts)$/,
+            include: path.dirname(entryPath),
+            use: {
+              loader: path.resolve(__dirname, './index.ts'),
+            },
+          },
+        ],
+      },
+    },
+    configOverrides,
+  );
+  const compiler = webpack(webpackConfig);
+
+  compiler.outputFileSystem = createFsFromVolume(new Volume());
+  compiler.outputFileSystem.join = path.join.bind(path);
+
+  return new Promise((resolve, reject) => {
+    compiler.run((err, stats) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      if (typeof stats === 'undefined') {
+        reject(new Error('"stats" from Webpack are not available, unknown error...'));
+        return;
+      }
+
+      const jsonStats = stats.toJson({ source: true });
+
+      if (stats.hasErrors()) {
+        reject(stats.toJson().errors![0]);
+        return;
+      }
+
+      if (!Array.isArray(jsonStats.modules)) {
+        reject(new Error(`"stats.toJson().modules" should be an array, this could be a compilation error...`));
+        return;
+      }
+
+      const entryModule = jsonStats.modules.find(module => module.nameForCondition === entryPath);
+
+      if (!entryModule) {
+        reject(new Error(`Failed to find a fixture in "stats.toJson().modules", this could be a compilation error...`));
+        return;
+      }
+
+      resolve(entryModule.source as string);
+    });
+  });
+}
+
+function fixLineEndings(value: string) {
+  return String(value).replace(/\r?\n/g, '\n').trim();
+}
+
+/**
+ * Test utility similar to "babel-plugin-tester".
+ *
+ * See https://webpack.js.org/contribute/writing-a-loader/#testing.
+ */
+function testFixture(fixtureName: string, configOverrides: webpack.Configuration = {}) {
+  it(`"${fixtureName}" fixture`, async () => {
+    const fixturePath = path.resolve(__dirname, '..', '__fixtures__', fixtureName);
+
+    const jsCodePath = path.resolve(fixturePath, 'code.js');
+    const tsCodePath = path.resolve(fixturePath, 'code.ts');
+    const tsxCodePath = path.resolve(fixturePath, 'code.tsx');
+
+    const jsOutputPath = path.resolve(fixturePath, 'output.js');
+    const tsOutputPath = path.resolve(fixturePath, 'output.ts');
+    const tsxOutputPath = path.resolve(fixturePath, 'output.tsx');
+
+    const inputPath = [
+      fs.existsSync(jsCodePath) && jsCodePath,
+      fs.existsSync(tsCodePath) && tsCodePath,
+      fs.existsSync(tsxCodePath) && tsxCodePath,
+    ].find(Boolean);
+    const outputPath = [
+      fs.existsSync(jsOutputPath) && jsOutputPath,
+      fs.existsSync(tsOutputPath) && tsOutputPath,
+      fs.existsSync(tsxOutputPath) && tsxOutputPath,
+    ].find(Boolean);
+
+    const errorPath = path.resolve(fixturePath, 'error.js');
```

Comment:
any particular reason why we don't support the same for `error` files ?

```
const jsErrorPath = path.resolve(fixturePath, 'error.js');
const tsErrorPath = path.resolve(fixturePath, 'error.ts');
const tsxErrorPath = path.resolve(fixturePath, 'error.tsx');

const errorPath = [
      fs.existsSync(jsErrorPath) && jsErrorPath,
      fs.existsSync(tsErrorPath) && tsErrorPath,
      fs.existsSync(tsxErrorPath) && tsxErrorPath,
    ].find(Boolean);

const expectedError = errorPath && require(errorPath);
```

with that only `packages/make-styles-webpack-loader/__fixtures__/error-syntax/code.js` needs to be a javascript file

Reply:
Code can be defined:
- by `.js` files when it is required (to avoid `@ts-ignore` as it tests)
- by `.ts`/`.tsx` in all other cases to grant type-safety

There is no sense to define errors with `.ts`/`.tsx`:
- it will require compilation (via `@babel/register`)
- error contains a single export via a string/regexp

Reply:
> it will require compilation (via @babel/register)

not true - see the screen-capture

your API as is, is not consistent for consumer -> thus my suggestions ( which work as expected ).

> - (to avoid @ts-ignore as it tests)

that can be mitigated with `ts-expect-error` pragma, which is the best practice for test anyways

Reply:
also your comment in one of the fixtures will become invalid once someone decides to check also `js` files

Reply:
> also your comment in one of the fixtures will become invalid once someone decides to check also js files

As Jest evaluates imports, yeah, I see...

> that can be mitigated with `ts-expect-error` pragma, which is the best practice for test anyways

Yeah, it's definitely better solution.

---

Summary for fd18b1f:
- `.js` support was removed, use `ts-expect-error`
- `.ts`, `.tsx`, `.txt` are supported for `code`
- `.ts`, `.tsx` are supported for `output`
- `.ts` are supported for errors

As we don't plan to publish this as util, this looks enough good for now.